compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -43,6 +43,23 @@ pub(super) enum RecoverQPath {
     No,
 }
 
+#[derive(Copy, Clone, PartialEq)]
+pub(super) enum RecoverReturnSign {
```

Comment:
I'd like to find a better name for this enum but I'm struggling to come up with anything.

Also, could you add doc comments to this enum, its variants and the `can_recover` method?